crates/relayer-cli/src/commands/query/client.rs

```diff
@@ -137,52 +137,49 @@ impl Runnable for QueryClientConsensusCmd {
-                                            .unwrap_or_else(exit_with_unrecoverable_error),
-                                    )
-                                },
-                            ),
-                        },
-                        IncludeProof::No,
-                    )
-                    .map(|(consensus_state, _)| consensus_state);
+        if let Some(cs_height) = self.consensus_height {
+            let consensus_height = Height::new(counterparty_chain.version(), cs_height)
+                .unwrap_or_else(exit_with_unrecoverable_error);
+
+            let res = chain
+                .query_consensus_state(
+                    QueryConsensusStateRequest {
+                        client_id: self.client_id.clone(),
+                        consensus_height,
+                        query_height: self.height.map_or(QueryHeight::Latest, |revision_height| {
+                            QueryHeight::Specific(
+                                Height::new(chain.id().version(), revision_height)
+                                    .unwrap_or_else(exit_with_unrecoverable_error),
+                            )
+                        }),
+                    },
+                    IncludeProof::No,
+                )
+                .map(|(consensus_state, _)| consensus_state);
+
+            match res {
+                Ok(cs) => Output::success(cs).exit(),
+                Err(e) => Output::error(e).exit(),
+            }
+        } else if self.heights_only {
+            let res = chain.query_consensus_state_heights(QueryConsensusStateHeightsRequest {
+                client_id: self.client_id.clone(),
+                pagination: Some(PageRequest::all()),
+            });
 
-                match res {
-                    Ok(cs) => Output::success(cs).exit(),
-                    Err(e) => Output::error(format!("{}", e)).exit(),
-                }
+            match res {
+                Ok(heights) => Output::success(heights).exit(),
+                Err(e) => Output::error(e).exit(),
             }
-            None => {
-                let res = chain.query_consensus_states(QueryConsensusStatesRequest {
-                    client_id: self.client_id.clone(),
-                    pagination: Some(PageRequest::all()),
-                });
-
-                match res {
-                    Ok(states) => {
-                        if self.heights_only {
-                            let heights: Vec<Height> = states.iter().map(|cs| cs.height).collect();
-                            Output::success(heights).exit()
-                        } else {
-                            Output::success(states).exit()
-                        }
-                    }
-                    Err(e) => Output::error(format!("{}", e)).exit(),
-                }
+        } else {
+            let res = chain.query_consensus_states(QueryConsensusStatesRequest {
```

Comment:
I think we should also remove the chain's `query_consensus_states()` as it is only used from the CLI. We can simplify the CLI to either list all consensus state heights or list one full consensus state at a given `consensus-height`.

Reply:
Agreed!

Reply:
Opened a PR targeting this branch here: https://github.com/informalsystems/hermes/pull/2990

Reply:
I approved #2990. Will you merge it here?

Reply:
Yep, as soon as CI passes :)(